jaeger-core/src/test/java/com/uber/jaeger/baggage/BaggageSetterTest.java

```diff
@@ -124,6 +124,31 @@ public void testUnsampledSpan() {
     assertNull(span.getLogs());
   }
 
+  @Test
+  public void testBaggageNullValueTolerated() {
+    when(mgr.getRestriction(SERVICE, KEY)).thenReturn(Restriction.of(true, 5));
+    final String value = null;
+    SpanContext ctx = setter.setBaggage(span, KEY, value);
+
+    assertBaggageLogs(span, KEY, null, false, false, false);
+    assertNull(ctx.getBaggageItem(KEY));
+  }
+
+  @Test
+  public void testBaggageNullRemoveValue() {
+    when(mgr.getRestriction(SERVICE, KEY)).thenReturn(Restriction.of(true, 5));
+    final String value = "value";
+    SpanContext ctx = setter.setBaggage(span, KEY, value);
+    Span child = (Span) tracer.buildSpan("some-operation").asChildOf(ctx).startManual();
+    ctx = setter.setBaggage(child, KEY, null);
```

Comment:
I would rather test this via span.setBaggage API, since it has additional null checks.